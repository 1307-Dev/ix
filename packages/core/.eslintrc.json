{
  "root": true,
  "extends": [
      "plugin:@stencil/recommended"
  ],
  "rules": {
    "react/jsx-no-bind": 0,
    "@stencil/async-methods": 0,
    "@stencil/strict-boolean-conditions": 0
  }
}
