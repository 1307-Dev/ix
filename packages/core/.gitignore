dist/
www/
loader/

*~
*.sw[mnpcod]
*.log
*.lock
*.tmp
*.tmp.*
log.txt
*.sublime-project
*.sublime-workspace

component-doc.json
.stencil/
.idea/
.vscode/
.sass-cache/
.versions/
node_modules/
$RECYCLE.BIN/

.DS_Store
Thumbs.db
UserInterfaceState.xcuserstate
.env
coverage
/test-results/
/playwright-report/
/playwright/.cache/


/ix-brand-theme
