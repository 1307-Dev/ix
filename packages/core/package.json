{
  "name": "@siemens/ix",
  "version": "0.0.0",
  "license": "Siemens Inner Source License v1.4",
  "description": "Stencil Component Starter",
  "main": "dist/index.cjs.js",
  "module": "dist/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "types": "dist/types/index.d.ts",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "unpkg": "dist/core-ui-core/core-ui-core.js",
  "files": [
    "dist/",
    "loader/",
    "src/**/*.md",
    "scss/"
  ],
  "scripts": {
    "build": "stencil build --docs --prod",
    "build:dev": "stencil build --docs",
    "lint": "eslint src/**/*{.ts,.tsx}",
    "start": "stencil build --dev --watch --serve",
    "dev": "stencil build --watch",
    "test": "stencil test --spec",
    "test.watch": "stencil test --spec --watchAll --no-cache",
    "test.coverage": "stencil test --spec --coverage",
    "test.e2e": "playwright test --reporter html",
    "test.e2e.list": "playwright test --reporter list",
    "host-root": "http-server ./ -a 127.0.0.1 -p 8080",
    "generate": "stencil generate",
    "postversion": "echo \"/*\n * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.\n */\nexport const version = '$npm_package_version'\" > ./src/env.ts",
    "generate.theme": "ts-node -P ./scripts/tsconfig.json ./scripts/generate-theme"
  },
  "dependencies": {
    "@siemens/core-ui-icons": "~1.1.16",
    "@stencil/core": "~2.15.0",
    "animejs": "~3.2.1",
    "hyperlist": "^1.0.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.24.1",
    "@popperjs/core": "~2.11.0",
    "@siemens/core-ui-icons": "^1.1.16",
    "@stencil/angular-output-target": "^0.4.0",
    "@stencil/eslint-plugin": "^0.4.0",
    "@stencil/postcss": "^2.0.0",
    "@stencil/react-output-target": "^0.3.1",
    "@stencil/sass": "~1.5.2",
    "@testing-library/dom": "^8.11.0",
    "@types/animejs": "^3.1.4",
    "@types/jest": "^26.0.18",
    "@types/mustache": "^4.1.1",
    "@types/puppeteer": "^5.4.0",
    "@typescript-eslint/eslint-plugin": "^4.33.0",
    "@typescript-eslint/parser": "^4.33.0",
    "animate.css": "~4.1.1",
    "autoprefixer": "10.4.5",
    "bootstrap": "~5.2.0",
    "cz-conventional-changelog": "^3.3.0",
    "eslint": "^7.32.0",
    "eslint-plugin-react": "^7.26.1",
    "fs-extra": "^10.0.0",
    "http-server": "^14.1.1",
    "jest": "^26.6.3",
    "jest-cli": "^26.6.3",
    "mustache": "^4.2.0",
    "puppeteer": "^10.0.0",
    "sass": "^1.54.0",
    "start-server-and-test": "^1.14.0",
    "ts-node": "^10.0.0",
    "typescript": "^4.2.4"
  },
  "peerDependencies": {
    "@popperjs/core": "^2.11.0",
    "bootstrap": "~5.2.0"
  },
  "publishConfig": {
    "registry": "***REMOVED***"
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-conventional-changelog"
    }
  }
}
