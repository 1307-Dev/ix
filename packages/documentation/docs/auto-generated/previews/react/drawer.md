<!-- Auto generated! Please edit here: siemens-ix/packages/react-test-app/src/preview-examples/drawer.tsx -->
```tsx
/*
 * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.
 */
import { IxButton, IxDrawer } from '@siemens/ix-react';
import React, { useState } from 'react';

export const DrawerFullHeight: React.FC = () => {
  const [show, setShow] = useState(false);

  return (
    <>
      <IxDrawer
        closeOnClickOutside={true}
        show={show}
        onDrawerClose={() => setShow(false)}
      >
        <span>Some content of drawer</span>
      </IxDrawer>
      <IxButton onClick={() => setShow(!show)}>Toggle drawer</IxButton>
    </>
  );
};
```
