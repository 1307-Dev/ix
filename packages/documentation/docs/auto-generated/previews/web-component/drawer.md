<!--
* COPYRIGHT (c) Siemens AG
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
-->
  
<!-- Auto generated! Please edit here: siemens-ix/packages/documentation/static/webcomponent-examples/drawer.html -->
```html

<ix-drawer>
  <span>Some content of drawer</span>
</ix-drawer>

<ix-button>Toggle drawer</ix-button>

<script>
  (async function () {
    await window.customElements.whenDefined('ix-drawer');

    const drawer = document.querySelector('ix-drawer');
    drawer.fullHeight = false;
    drawer.closeOnClickOutside = true;

    const button = document.querySelector('ix-button');
    button.addEventListener('click', () => {
      drawer.show = !drawer.show;
    });
  })();
</script>
```
