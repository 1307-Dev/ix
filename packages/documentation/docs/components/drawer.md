import LivePreview from '@site/src/components/LivePreview';
import ComponentApi from '@site/src/components/ComponentApi';

# Drawer

<LivePreview name="drawer-full-height" height="24rem"></LivePreview>

## Auto height

<LivePreview name="drawer" height="24rem"></LivePreview>

<ComponentApi name="cui-drawer"></ComponentApi>
<ComponentApi name="cw-drawer"></ComponentApi>
