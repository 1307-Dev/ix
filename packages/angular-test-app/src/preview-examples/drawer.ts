/*
 * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.
 */
import { Component } from '@angular/core';

@Component({
  selector: 'app-drawer',
  template: `
    <ix-drawer [closeOnClickOutside]="true" [show]="show">
      <span>Some content of drawer</span>
    </ix-drawer>

    <ix-button (click)="show = !show">Toggle drawer</ix-button>
  `,
})
export class Drawer {
  show = false;
}
