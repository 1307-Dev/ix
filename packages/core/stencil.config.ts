import { angularOutputTarget } from '@stencil/angular-output-target';
import { Config } from '@stencil/core';
import { postcss } from '@stencil/postcss';
import { reactOutputTarget } from '@stencil/react-output-target';
import { sass } from '@stencil/sass';
import autoprefixer from 'autoprefixer';
export const config: Config = {
  bundles: [
    {
      components: ['cw-animated-tabs', 'cw-animated-tab'],
    },
    {
      components: ['cw-application-header'],
    },
    {
      components: ['cw-basic-navigation'],
    },
    {
      components: ['cw-blind'],
    },
    {
      components: ['cw-breadcrumb', 'cw-breadcrumb-item'],
    },
    {
      components: ['cw-button'],
    },
    {
      components: ['cw-category-filter'],
    },
    {
      components: ['cw-chip'],
    },
    {
      components: ['cw-counter-pill'],
    },
    {
      components: ['cw-drawer'],
    },
    {
      components: ['cw-dropdown', 'cw-dropdown-item'],
    },
    {
      components: ['cw-event-list', 'cw-event-list-item'],
    },
    {
      components: ['cw-expanding-search'],
    },
    {
      components: ['cw-filter-chip'],
    },
    {
      components: ['cw-flip-tile', 'cw-flip-tile-content'],
    },
    {
      components: ['cw-group', 'cw-group-dropdown-item', 'cw-group-item'],
    },
    {
      components: ['cw-icon'],
    },
    {
      components: ['cw-icon-button'],
    },
    {
      components: ['cw-input-group'],
    },
    {
      components: ['cw-map-navigation', 'cw-map-navigation-overlay'],
    },
    {
      components: [
        'cw-menu',
        'cw-menu-item',
        'cw-menu-about',
        'cw-menu-about-item',
        'cw-menu-about-news',
        'cw-menu-avatar',
        'cw-menu-avatar-item',
        'cw-menu-settings',
        'cw-menu-settings-item',
      ],
    },
    {
      components: ['cw-message-bar'],
    },
    {
      components: ['cw-modal', 'cw-modal-container'],
    },
    {
      components: ['cw-pill'],
    },
    {
      components: ['cw-select', 'cw-select-item'],
    },
    {
      components: ['cw-spinner'],
    },
    {
      components: ['cw-split-button', 'cw-split-button-item'],
    },
    {
      components: ['cw-tile'],
    },
    {
      components: ['cw-toast', 'cw-toast-container'],
    },
    {
      components: ['cw-toggle'],
    },
    {
      components: ['cw-tree', 'cw-tree-item'],
    },
    {
      components: ['cw-upload'],
    },
  ],
  extras: {
    appendChildSlotFix: true,
    slotChildNodesFix: true,
  },
  namespace: 'core-ui-core',
  globalStyle: './scss/v7/core-ui.scss',
  // globalStyle: 'scss/core-ui.scss',
  minifyCss: false,
  plugins: [
    sass({
      includePaths: ['./scss/v7', './scss'],
    }),
    postcss({
      plugins: [autoprefixer()],
    }),
  ],
  outputTargets: [
    angularOutputTarget({
      componentCorePackage: '@siemens/ix',
      directivesProxyFile: '../angular/src/components.ts',
      directivesArrayFile: '../angular/src/declare-components.ts',
      excludeComponents: ['my-component'],
    }),
    reactOutputTarget({
      componentCorePackage: '@siemens/ix',
      proxiesFile: '../react/src/components.ts',
      includeDefineCustomElements: true,
      excludeComponents: ['my-component'],
    }),
    {
      type: 'dist',
      esmLoaderPath: '../loader',
      copy: [
        {
          src: '../scss/fonts',
          dest: '../core-ui-core/fonts',
        },
      ],
    },
    {
      type: 'dist-custom-elements',
    },
    {
      type: 'docs-json',
      file: 'component-doc.json',
    },
    {
      type: 'docs-readme',
      footer: '',
      dependencies: false,
    },
    {
      type: 'www',
      serviceWorker: null, // disable service workers
      copy: [
        {
          src: './../../../node_modules/@siemens/core-ui-icons/dist',
          dest: 'build/core-ui-webfont',
        },
        {
          src: '../scss/fonts',
          dest: 'build/fonts',
        },
      ],
    },
  ],
};
