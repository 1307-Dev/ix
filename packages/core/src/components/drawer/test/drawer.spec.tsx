/*
 * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.
 */

import { newSpecPage } from '@stencil/core/testing';
import { fireEvent } from '@testing-library/dom';
import { CwDrawer } from '../drawer';

describe('ix-drawer', () => {
  let page: any;
  let drawer: any;
  let container: HTMLDivElement;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [CwDrawer],
      html: `<ix-drawer></ix-drawer>`,
    });

    drawer = document.querySelector('ix-drawer');
    container = document.querySelector('[data-testid="container"]');
  });

  it('renders', () => {
    expect(page.root).toMatchSnapshot();
  });

  it('opens the drawer', async () => {
    drawer.show = true;
    await page.waitForChanges();

    expect(drawer.show).toBeTruthy();

    await page.waitForChanges();
    expect(drawer.innerHTML).toContain('toggle');
  });

  it('closes the drawer', async () => {
    drawer.show = true;
    await page.waitForChanges();

    const closeButton = page.doc.querySelector('[data-testid="close-button"]');
    fireEvent.click(closeButton);
    await page.waitForChanges();
    expect(drawer.show).toBeFalsy();
    expect(drawer.innerHTML).not.toContain('toggle');
  });

  it('drawer is displayed at full height, if fullHeight is set to true', async () => {
    drawer.toggleDrawer();
    await page.waitForChanges();

    drawer.fullHeight = true;
    await page.waitForChanges();

    expect(container.classList.contains('full-height')).toBeTruthy();
  });

  it('drawer is NOT displayed at full height, if fullHeight is set to false', async () => {
    drawer.toggleDrawer();
    await page.waitForChanges();

    drawer.fullHeight = false;
    await page.waitForChanges();

    expect(container.classList.contains('full-height')).toBeFalsy();
  });

  it('emits an event, when show changed', async () => {
    const mockCallback = jest.fn();
    window.addEventListener('open', mockCallback);

    drawer.toggleDrawer();
    await page.waitForChanges();

    const closeButton = page.doc.querySelector('[data-testid="close-button"]');
    fireEvent.click(closeButton);

    window.removeEventListener('open', mockCallback);
    expect(mockCallback).toHaveBeenCalled();
  });
});
