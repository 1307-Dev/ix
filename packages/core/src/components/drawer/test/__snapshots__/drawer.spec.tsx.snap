// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cw-drawer renders 1`] = `
<cw-drawer>
  <!---->
  <div class="d-none drawer-container" data-testid="container" id="div-container" style="width: 16rem; min-width: 16rem; max-width: 28rem;">
    <div class="header">
      <div class="header-content"></div>
      <cw-icon-button data-testid="close-button" icon="close" ghost="" size="24"></cw-icon-button>
    </div>
    <div class="content"></div>
  </div>
</cw-drawer>
`;
